src/OpenFeature.DependencyInjection/Internal/FeatureLifecycleManager.cs

```diff
@@ -0,0 +1,37 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace OpenFeature.Internal;
+
+internal sealed class FeatureLifecycleManager : IFeatureLifecycleManager
+{
+    private readonly Api _featureApi;
+    private readonly IServiceProvider _serviceProvider;
+    private readonly ILogger<FeatureLifecycleManager> _logger;
+
+    public FeatureLifecycleManager(Api featureApi, IServiceProvider serviceProvider, ILogger<FeatureLifecycleManager> logger)
+    {
+        _featureApi = featureApi;
+        _serviceProvider = serviceProvider;
+        _logger = logger;
+    }
+
+    /// <inheritdoc />
+    public async ValueTask EnsureInitializedAsync(CancellationToken cancellationToken = default)
+    {
+        _logger.LogInformation("Starting initialization of the feature provider");
```

Comment:
@askpt Could you please clarify the need for using the source generator for logging?

Reply:
Thanks for the suggestion, but I’d prefer to stick with the standard logging in the HostedService. I don't think it's necessary here if the main reason is performance or typing. We're not using any generated code or partial methods for logging, and the standard approach is more familiar and widely used. Given the current scope, switching would add complexity without significant benefit.

Reply:
I'd prefer it for consistency since it's what's used elsewhere in the SDK, but it's not a blocker for me, personally.

Reply:
I agree with @toddbaert. Especially since this brings some high-performance logging to the table, if you need help, I can make the changes in the PR.

Reply:
Yes, please.

Reply:
Changed 👍